notebooks/04_model_select_and_optimize/azureml_hyperdrive_tuning_svd.ipynb

```diff
@@ -0,0 +1,731 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>Copyright (c) Microsoft Corporation. All rights reserved.<br>\n",
+    "Licensed under the MIT License.</i>\n",
+    "<br><br>\n",
+    "# SVD Hyperparameter Tuning with Azure Machine Learning"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this notebook, we show how to tune the hyperparameters of a matrix factorization algorithm by utilizing **Azure Machine Learning service** ([AzureML](https://azure.microsoft.com/en-us/services/machine-learning-service/)) in the context of movie recommendations. To use AzureML you will need an Azure subscription. We use the SVD algorithm from the Surprise library.\n",
+    "\n",
+    "We present the overall process of utilizing AzureML by demonstrating some key steps while avoiding too much detail. \n",
+    "\n",
+    "For more details about the **SVD** algorithm:\n",
+    "* [Surprise SVD deep-dive notebook](../02_model/surprise_svd_deep_dive.ipynb)\n",
+    "* [Original paper](http://papers.nips.cc/paper/3208-probabilistic-matrix-factorization.pdf)\n",
+    "* [Surprise homepage](https://surprise.readthedocs.io/en/stable/)\n",
+    "  \n",
+    "Regarding **AzureML**, please refer to:\n",
+    "* [Quickstart notebook](https://docs.microsoft.com/en-us/azure/machine-learning/service/quickstart-create-workspace-with-python)\n",
+    "* [Hyperdrive](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-tune-hyperparameters)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "System version: 3.6.8 |Anaconda, Inc.| (default, Feb 11 2019, 15:03:47) [MSC v.1915 64 bit (AMD64)]\n",
+      "Surprise version: 1.0.6\n",
+      "Azure ML SDK Version: 1.0.10\n"
+     ]
+    }
+   ],
+   "source": [
+    "import sys\n",
+    "sys.path.append(\"../../\")\n",
+    "import time\n",
+    "import os\n",
+    "import surprise\n",
+    "import papermill as pm\n",
+    "import pandas as pd\n",
+    "import shutil\n",
+    "from reco_utils.dataset import movielens\n",
+    "from reco_utils.dataset.python_splitters import python_random_split\n",
+    "\n",
+    "print(\"System version: {}\".format(sys.version))\n",
+    "print(\"Surprise version: {}\".format(surprise.__version__))\n",
+    "\n",
+    "import azureml as aml\n",
+    "import azureml.widgets\n",
+    "import azureml.train.hyperdrive as hd\n",
+    "\n",
+    "print(\"Azure ML SDK Version:\", aml.core.VERSION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prerequisite\n",
```

Comment:
Is it better to organize the notebook structure following our template which is used by other notebooks?

Reply:
What is the template?